### PR TITLE
Fixed bugs with save oAuth tokens and API docs

### PR DESCRIPTION
**Pull request by oudeismetis**
This connects to istrategylabs/franklin#93

### CODE REVIEW

File: franklin/core/views.py [hunk 1]
```diff
@@ -1,12 +1,88 @@
...
 
 
+class ConvertTokenView(social_oauth_views.ConvertTokenView, APIView):
+    """ Login user """
+    permission_classes = (AllowAny,)
+
+    def post(self, request, *args, **kwargs):
+        """ 
+        Exchange social token for franklin oAuth access token
+        ---
```
Comment (oudeismetis) [inline franklin/core/views.py]: django-rest-swagger wasn't auto detecting the endpoints created by django-rest-framework-social-oauth2. For now, I'm going to extend each of their views so we have properly documented endpoints. For now it'll just be this endpoint and this one is important for dashboard integration. 

I'm going to look into fixing this in django-rest-framework-social-oauth2 as that library is supposed to play nicely with django-rest-framework.

File: franklin/core/views.py [hunk 2]
```diff
@@ -1,12 +1,88 @@
...
+    # TODO django-rest-framework-social-oauth2 has a bug in it (11/2015)
+    # It doesn't save the access_token, so we do it here manually
+    request = kwargs.get('request', None)
+    if user and request:
+        access_token = request.get('token', None)
+        social = user.social_auth.get(provider='github')
+        social.extra_data['access_token'] = access_token
+        social.save()
+        return { 'is_new': False, 'user': user }
+    return { 'is_new': True, 'user': None }
```
Comment (oudeismetis) [inline franklin/core/views.py]: Another issue with django-rest-framework-social-oauth2. It's not saving the oAuth token we get from Github social signin. As with above, I'm going to attempt to fix and PR the fix into that project. For now, I've injected a step into python-social-auth that manually pulls the token off of the request and saves it into the users social model after it's been created and associated.

File: franklin/github/views.py [hunk 3]
```diff
@@ -23,11 +23,11 @@
...
     url = github_base + 'repos/' + site.owner.name + '/' + site.name + '/contents/.franklin.yml'
     #TODO - This will fetch the file from the default master branch
-    #social = user.social_auth.get(provider='github')
-    # token = social.extra_data['access_token']
+    social = user.social_auth.get(provider='github')
+    token = social.extra_data['access_token']
     headers = {
                 'content-type': 'application/json',
-                'Authorization': 'token ' + os.environ['GITHUB_OAUTH'] 
+                'Authorization': 'token ' + token
```
Comment (oudeismetis) [inline franklin/github/views.py]: social tokens work again. No more hard coding in vars :boom:

**Comment by oudeismetis** [comment]
This PR is ready for thumbs

**Comment by JulianGindi** [comment]
:+1:

**Pull request merged**